### ISSUE TITLE
Zstandard is in the source code, but does not exist any doc about it

### ISSUE THREAD

**Issue by jefer94** [oven-sh/bun#7418]
### What is the type of issue?

Documentation is missing

### What is the issue?

There have Zstandard referenced in many files, but it's not documented yet
![image](https://github.com/oven-sh/bun/assets/3018142/b8f5d247-d234-4268-b518-6d5afc40f9df)

This should be useful to optimize the deployments in the frontend 

### Where did you find it?

https://bun.sh/guides
https://bun.sh/docs

**Comment by Jarred-Sumner** [comment]
indeed, it is only currently used with `bun build --compile --sourcemap` to compress the sourcemaps

**Comment by jefer94** [comment]
Should be great to have a module to use the same implementation within Javascript, and it should be a replacement to Brotli until this is implemented, the good part of use Brotli or Zstandard if that I have implemented both within a cache in Redis, but maybe I should look for another compression algorithm, all packages related to them looks abandoned, whatever currently Zstd is experimental in Blink browsers yet

**Comment by L3P3** [comment]
I would have prefered namespaces but as we already have `Bun.gzipSync`, there could just as well be a `Bun.zstdSync`. Maybe this will be my first contribution, never worked with zig however. :sweat_smile:
Even though it could be handy, I would not overestimate its usage. In productive environments, I highly recommend using nginx as a reverse proxy and bun behind it. Or are there benchmarks speaking against it?

**Comment by bompus** [comment]
I saw mention of `Bun.zstdSync`, so figured I would mentiond that node has a [PR](https://github.com/nodejs/node/pull/52100) pending to add this, and it'd be fantastic to have it in Bun natively, rather than needing to import a 3rd-party npm n-api implementation for this functionality.

If the compressor/decompressor logic is already in source, what would it take to expose it to javascript-land?

**Comment by L3P3** [comment]
Not much, mostly wrapper code, tests (!) and updating documentation.
But I think before we should clarify if we keep n compression algorithms directly in Bun namespace or do it better for newly added ones.

**Comment by jefer94** [comment]
Right now `zstd` is supported for the browsers, I am not affected by this because I use Next in Vercel, and Vercel uses Node

**Comment by L3P3** [comment]
zstd is now also in node:zlib, so it will be usable anyways.

**Comment by Jarred-Sumner** [comment]
Zstd decompression and compression is exposed as of [Bun v1.2.17](https://bun.sh/blog/bun-v1.2.17#zstandard-zstd-support-in-node-zlib)
